File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        
        int curx = nums[0] ;
        int maxv = nums[0] ; 
        for(int i=1 ; i < nums.size() ; i++){
            int n = nums[i] ;
            curx = max(curx+n , n) ; 
            maxv = max(curx , maxv) ;
            
        }
        return maxv ;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        
        int curSubv = nums[0] ; 
        int maxSubv = nums[0] ;
        for(int i=01 ; i < nums.length ; i++){
            
            int n = nums[i] ; 
            curSubv = Math.max(curSubv+n ,  n  ) ; 
            maxSubv = Math.max( curSubv , maxSubv ) ; 
        }
        return maxSubv ; 
        
    }
}